src/DssSpell.t.sol

```diff
@@ -614,369 +604,426 @@ contract DssSpellTest is DssSpellTestBase {
+        assertEq(rwa015AInputConduitUrn.gem(), addr.addr("USDC"),           "input-conduit-urn-gem-not-match");
+
+        assertEq(rwa015AInputConduitJar.psm(), addr.addr("MCD_PSM_USDC_A"), "input-conduit-jar-psm-not-match");
+        assertEq(rwa015AInputConduitJar.to(),  address(rwa015AJar),         "input-conduit-jar-to-not-match");
+        assertEq(rwa015AInputConduitJar.dai(), addr.addr("MCD_DAI"),        "input-conduit-jar-dai-not-match");
+        assertEq(rwa015AInputConduitJar.gem(), addr.addr("USDC"),           "input-conduit-jar-gem-not-match");
     }
 
-    function testRWA014_INTEGRATION_CONDUITS_SETUP() public {
+    function testRWA015_INTEGRATION_CONDUITS_SETUP() public {
         _vote(address(spell));
         _scheduleWaitAndCast(address(spell));
         assertTrue(spell.done());
 
-        assertEq(rwaconduitout_014.wards(ESM), 1, "OutputConduit/ward-esm-not-set");
-        assertEq(rwaconduitout_014.can(pauseProxy), 1, "OutputConduit/pause-proxy-not-hope");
-        assertEq(rwaconduitout_014.can(RWA014_A_OPERATOR), 1, "OutputConduit/operator-not-hope");
-        assertEq(rwaconduitout_014.may(pauseProxy), 1, "OutputConduit/pause-proxy-not-mate");
-        assertEq(rwaconduitout_014.may(RWA014_A_OPERATOR), 1, "OutputConduit/operator-not-mate");
-        assertEq(rwaconduitout_014.quitTo(), address(rwaurn_014), "OutputConduit/quit-to-not-urn");     
-        assertEq(rwaconduitout_014.bud(RWA014_A_COINBASE_CUSTODY), 1, "OutputConduit/destination-address-not-whitelisted-for-pick");
-
-        assertEq(rwaconduitinurn_014.wards(ESM), 1, "InputConduitUrn/ward-esm-not-set");
-        assertEq(rwaconduitinurn_014.may(pauseProxy), 1, "InputConduitUrn/pause-proxy-not-mate");
-        assertEq(rwaconduitinurn_014.may(RWA014_A_OPERATOR), 1, "InputConduitUrn/operator-not-mate");
-        assertEq(rwaconduitinurn_014.quitTo(), RWA014_A_COINBASE_CUSTODY, "InputConduitUrn/quit-to-not-set");
-
-        assertEq(rwaconduitinjar_014.wards(ESM), 1, "InputConduitJar/ward-esm-not-set");
-        assertEq(rwaconduitinjar_014.may(pauseProxy), 1, "InputConduitJar/pause-proxy-not-mate");
-        assertEq(rwaconduitinjar_014.may(RWA014_A_OPERATOR), 1, "InputConduitJar/operator-not-mate");
-        assertEq(rwaconduitinjar_014.quitTo(), RWA014_A_COINBASE_CUSTODY, "InputConduitJar/quit-to-not-set");
-
-        assertEq(rwajoin_014.wards(address(rwaurn_014)), 1, "Join/ward-urn-not-set");
-        assertEq(rwajoin_014.wards(ESM), 1, "Join/ward-esm-not-set");
-
-        assertEq(rwaurn_014.wards(ESM), 1, "Urn/ward-esm-not-set");
-        assertEq(rwaurn_014.can(pauseProxy), 1, "Urn/pause-proxy-not-hoped");
-        assertEq(rwaurn_014.can(RWA014_A_OPERATOR), 1, "Urn/operator-not-hoped");
+        assertEq(rwa015AOutputConduit.wards(pauseProxy),      1, "OutputConduit/ward-pause-proxy-not-set");
+        assertEq(rwa015AOutputConduit.wards(address(esm)),    1, "OutputConduit/ward-esm-not-set");
```

Comment:
BTW RWA014 uses address ESM                                = addr.addr("MCD_ESM"); to load up the ESM.

You can use MCD_ESM as the varname here and could have used it without the address() casts, which would be simpler.

Not requesting a change, just letting you know

In future we should either have ESM checks in the auth type section (as it is done here) or all together at the bottom of this check, and it should match the main spell (i.e. main spell relies ESM all together but checks it as part of permissions here) - it is just cleaner.

Not necessary for now though as we are already over time

Reply:
there's already a pattern , i don't want to introduce big diffs this late into the process unless there's a serious ground (e.g. security concern)

Reply:
Whatever pattern you're following does not match RWA014 onboarding from last week, where ESM auth checks were added to the RWA test.

Resolving for now but let's not change from the archive pattern (RWA014 in this case) unless there is a good reason.

Changing stuff from archive pattern for no reason is a good enough justification for it to be changed back IMO.

Resolving for now but needs fixing in the future.